### PR TITLE
feat: Update Angular template to version 19

### PR DESCRIPTION
**Pull request by alan-agius4**
This prerelease version 19 includes several developer experience (DX) improvements:
- Introduces a new server routing API
- Portable with no Node.js dependencies
- Supports a "neutral" output format

Fixes #0000

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [ ] Included
  - [x] Not necessary because: covered by C3 e2e tests
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): <https://github.com/cloudflare/cloudflare-docs/pull/>...
  - [x] Not necessary because: no change to developer docs for Angular

### CODE REVIEW

**Comment by petebacondarwin** [comment]
Cool! Thanks for putting this together @alan-agius4 - I'll try running the e2e tests on our fork to check it is all good.

File: packages/create-cloudflare/templates/angular/c3.ts [hunk 1]
```diff
@@ -93,8 +104,7 @@ const config: TemplateConfig = {
 		scripts: {
 			start: `${npm} run build && wrangler pages dev dist/cloudflare ${await compatDateFlag()} --experimental-local`,
 			build: `ng build && ${npm} run process`,
-			process:
-				"node ./tools/copy-files.mjs && node ./tools/alter-polyfills.mjs",
```
Comment (petebacondarwin) [inline packages/create-cloudflare/templates/angular/c3.ts:97]: 🎉

File: packages/create-cloudflare/templates/angular/c3.ts [hunk 2]
```diff
@@ -50,12 +50,21 @@ async function updateAppCode() {
...
+	);
+	writeFile(resolve(appServerRoutesPath), newAppRoutes);
+	s.stop(`${brandColor(`updated`)} ${dim(appServerRoutesPath)}`);
+
 	// Remove unwanted dependencies
 	s.start(`Updating package.json`);
 	const packageJsonPath = resolve("package.json");
 	const packageManifest = readJSON(packageJsonPath);
 
-	delete packageManifest["dependencies"]["@angular/ssr"];
```
Comment (petebacondarwin) [inline packages/create-cloudflare/templates/angular/c3.ts:58]: 🎉

File: .changeset/cool-experts-punch.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"create-cloudflare": minor
+---
+
+Update Angular template to version 19
```
Comment (petebacondarwin) [inline .changeset/cool-experts-punch.md:5]:
```suggestion
feat: update Angular template to version 19
```

**Review by petebacondarwin — APPROVED** [review]
E2E tests pass on pnpm and Mac for me locally!

Do we want to wait for Angular 19 to come out of pre-release @alan-agius4 ?

**Comment by alan-agius4** [comment]
> Do we want to wait for Angular 19 to come out of pre-release @alan-agius4 ?

SGTM, the final is schedule in mid November, and RC later this month.

**Comment by petebacondarwin** [comment]
> > Do we want to wait for Angular 19 to come out of pre-release @alan-agius4 ?
> 
> SGTM, the final is schedule in mid November, and RC later this month.

How about we bump this when the RC is out to encourage testing of the new version?

**Comment by alan-agius4** [comment]
Sure, thanks

**Comment by alan-agius4** [comment]
@petebacondarwin, version 19 has been released as stable.

**Review by jakkal6 — COMMENTED** [review]
[](https://apps.apple.com/za/app/calendly-mobile/id1451094657)

**Pull request merged**